### PR TITLE
Add tqdm progress bar to training loop method. 

### LINKED ISSUE #25: [Python] Progress Bar (tqdm) for Training
### Description
Current training progress is reported via standard `print()` statements every 10 epochs. A proper progress bar improves the developer experience significantly.

### Implementation Details
- Add `tqdm` to dependencies.
- In `api.py`, wrap the training loop in `tqdm(range(epochs))`.
- Update the progress bar description with the current loss value: `pbar.set_description(f"Loss: {current_loss:.4f}")`.

### Acceptance Criteria
- `Model.train()` shows a live-updating progress bar in the terminal or Jupyter notebook.

### PR DESCRIPTION
**Pull request by DhineshPonnarasan**
Add tqdm progress bar to training loop method.
---

Fixes
---
Closes #25 

Type of Change
---
- [ ]  Bug fix
- [ ]  New feature
- [ ]  Documentation / Refactor
- [ ]  Math / Logic correction

Description
---
This PR adds a tqdm-based progress bar to the training loop to improve visibility during model training.
The progress bar displays epoch-level progress and dynamically updates with the current loss value.
No existing training logic or public APIs were modified. 

How Has This Been Tested?
---

- [ ] Unit Tests: Verified behavior using controlled training runs and loss history checks.
- [ ] Manual Testing: Verified logic through a dummy training loop and visually confirmed progress bar updates.
- [ ] Integration: Confirmed Rust core and Python API communicate correctly during training.

Screenshots / Logs
---

Screenshot 2026-01-01 064014

Contribution Context
---

- [x]  I am contributing through the SWOC program.

### CODE REVIEW

**Comment by DhineshPonnarasan** [comment]
Hi Aamod Kumar, 
Implemented per-epoch training with tqdm on the feature/add-tqdm-progress-bar branch.
The Python layer now iterates over epochs and calls train_one_epoch, while the Rust core exposes a new train_one_epoch method that returns the per-epoch loss.
No unrelated changes were made to Cargo.toml; updates to lib.rs are strictly to expose the new method.


PS C:\Users\diino\Documents\SWOC 26\etna> cd "C:\Users\diino\Documents\SWOC 26\etna-tqdm" ; .\.venv\Scripts\python.exe progress_bar_live_test.py
🔮 Auto-Detected Task: Classification (Target 'target')
⚙️  Preprocessing data...
🚀 Initializing Rust Core [In: 2, Out: 2]...
🔥 Training started...
Training: 100%|██████████████████████████████████████████████████████████| 3/3 [00:00<?, ?epoch/s] 
✅ Training complete!
✅ Progress bar acceptance test passed: per-epoch losses recorded.

**Review by Aamod007 — APPROVED** [review]
## Review: Approve

Hi @DhineshPonnarasan, excellent work on this implementation.

Adding `train_one_epoch()` to the Rust backend was the right approach, the progress bar will now show real time updates during training. Clean implementation with proper Python bindings.

This PR is ready to merge.

---

Regards,
**Aamod Kumar**
Mentor, SWOC '26

**Comment by DhineshPonnarasan** [comment]
Hi Soham,

I checked the branch against main, and test_metrics.py already matches the main branch with no differences. There are no pending changes related to that file.

The only remaining item was an untracked verify_implementation.py file, which has not been committed. Based on this, it looks like either the branch never had changes to test_metrics.py, or any earlier changes were already reverted in a previous commit.

The branch is now clean with respect to test_metrics.py and fully aligned with main, which should address the test failures that were reported earlier.

Please let me know if you’d like me to double-check anything else.

Regards,
Dhinesh Ponnarasan

**Comment by DhineshPonnarasan** [comment]
I have merged the latest main into my feature branch, resolved the merge successfully and pushed the updated changes.
The PR is now conflict-free and up to date. Please let me know if anything else is needed from my side.
